include/transport/priority_queue.h

```diff
@@ -171,7 +171,13 @@ namespace qtransport {
         }
 
         bool empty() const {
-            return size() == 0;
+            for (size_t i = 0; i < _queue.size(); i++) {
+                if (_queue[i] && !_queue[i]->empty()) {
+                    return false;
+                }
+            }
+
+            return true;
         }
```

Comment:
Alternatively, we could let STL do it for us:
```suggestion
        bool empty() const {
            return std::all_of(_queue.begin(), _queue.end(), [&](const auto& elem) { return elem->empty(); });
        }
```